src/Main.elm

```diff
@@ -49,126 +49,200 @@ type DecimalStatus
     | NoDecimal
 
 
-nextAction : Model -> Maybe Action
-nextAction model =
-    case model.actionB of
-        Just Equals ->
-            Nothing
+type ActionType
+    = AddSubtractType
+    | MultiplyDivideType
+    | EqualsType
 
-        action ->
-            action
 
 
-saveResults : Model -> Float -> Model
-saveResults model results =
-    { model
-        | display = showResults <| results
-        , valueA = results
-        , actionA = nextAction model
-        , valueB = 0
-        , actionB = Nothing
-        , displayStatus = ShowingResults
-    }
+-- Update
 
 
-executeAdd : Model -> Model
-executeAdd model =
-    saveResults
-        model
-        (model.valueA + model.valueB)
+type Msg
+    = UpdateDisplay CalButton
+    | Saveinput Action
+    | AllClear
 
 
-executeSubtract : Model -> Model
-executeSubtract model =
-    saveResults
-        model
-        (model.valueA - model.valueB)
+saveAction : Model -> Action -> Model
+saveAction model newAction =
+    case ( model.actionA, model.actionB ) of
+        ( Just Equals, Nothing ) ->
+            saveActionA model newAction
 
+        ( Nothing, Nothing ) ->
+            saveActionA model newAction
 
-executeMultiply : Model -> Model
-executeMultiply model =
-    saveResults
-        model
-        (model.valueA * model.valueB)
+        ( Just a, Nothing ) ->
+            checkActionAandNewAction model a newAction
 
+        ( Just a, Just b ) ->
+            checkActionBandNewAction model a b newAction
 
-executeDivide : Model -> Model
-executeDivide model =
-    saveResults
-        model
-        (model.valueA / model.valueB)
+        ( Nothing, Just b ) ->
+            model
```

Comment:
This code path indicates a button press that would do nothing if it was ever hit, right? That's a smell that indicates something is off in the model. My guess is that it's related to how you're storing the actions, which allows for action b to be set without action a ever being set.

I would address this in a future PR though.